library/core/src/num/dec2flt/number.rs

```diff
@@ -59,30 +60,32 @@ impl Number {
         // require setting it by changing the global state (like the control word of the x87 FPU).
         let _cw = set_precision::<F>();
 
-        if self.is_fast_path::<F>() {
-            let mut value = if self.exponent <= F::MAX_EXPONENT_FAST_PATH {
-                // normal fast path
-                let value = F::from_u64(self.mantissa);
-                if self.exponent < 0 {
-                    value / F::pow10_fast_path((-self.exponent) as _)
-                } else {
-                    value * F::pow10_fast_path(self.exponent as _)
-                }
+        if !self.can_use_fast_path::<F>() {
+            return None;
+        }
+
+        let mut value = if self.exponent <= F::MAX_EXPONENT_FAST_PATH {
+            // normal fast path
+            let value = F::from_u64(self.mantissa);
+            if self.exponent < 0 {
+                value / F::pow10_fast_path((-self.exponent) as _)
             } else {
-                // disguised fast path
-                let shift = self.exponent - F::MAX_EXPONENT_FAST_PATH;
-                let mantissa = self.mantissa.checked_mul(INT_POW10[shift as usize])?;
-                if mantissa > F::MAX_MANTISSA_FAST_PATH {
-                    return None;
-                }
-                F::from_u64(mantissa) * F::pow10_fast_path(F::MAX_EXPONENT_FAST_PATH as _)
-            };
-            if self.negative {
-                value = -value;
+                value * F::pow10_fast_path(self.exponent as _)
             }
-            Some(value)
         } else {
-            None
+            // disguised fast path
+            let shift = self.exponent - F::MAX_EXPONENT_FAST_PATH;
+            let mantissa = self.mantissa.checked_mul(INT_POW10[shift as usize])?;
+            if mantissa > F::MAX_MANTISSA_FAST_PATH {
+                return None;
+            }
+            F::from_u64(mantissa) * F::pow10_fast_path(F::MAX_EXPONENT_FAST_PATH as _)
+        };
+
+        if self.negative {
+            value = -value;
         }
+
+        Some(value)
```

Comment:
```suggestion
        if self.negative {
            Some(-value)
        } else {
            Some(value)
        }
```

is there a reason why it is better to keep value as `mut`?

Reply:
Thanks!

This is totally not done yet so don't look too close (I'm waiting on https://github.com/rust-lang/rust/pull/127510 so I can test more completely).

Reply:
apologies for looking too close before you were finished XD

Reply:
I'll ping you to take a look once it is ready, especially if you know your float stuff :)

Reply:
alas I do not know much float stuff, I mostly just enjoy lurking the rust pull requests